backend/endpoints/models/user_models.go

```diff
@@ -32,16 +32,16 @@ func (u *User) IsValidEmail() bool {
 
 // UserExists just checks if a user has a valid password
 func (u *User) UserExists(personRepo repositories.PersonRepository) bool {
-	hashedPassword := u.hashPassword()
+	hashedPassword := u.HashPassword()
 
 	return personRepo.PersonExists(repositories.Person{
 		Email:    u.Email,
 		Password: hashedPassword,
 	})
 }
 
-// hashPassword hashes a user's password
-func (u *User) hashPassword() string {
+// HashPassword hashes a user's password
+func (u *User) HashPassword() string {
```

Comment:
I kind of don't like the idea of exporting this just so it can be used by tests, my main gripe with the idea is that it suddenly allows for someone to completely change the hash algorithm and tests would still pass, ie redefining `HashPassword` to
```go
func (u *User) HashPassword() string {
    return u.Password
}
```
would still pass all the tests but would still be a problematic change :(

Reply:
Hmm that makes a lot of sense.

With the current implementation, it is necessary that the tests know how passwords are hashed to mock the person repository. The only way of avoiding exporting `hashPassword` I can think of is adding a hash function to `IPersonRepository`. Thoughts?

To that last remark, I think it's also worth considering if it is the responsibility of these tests to check if the hashing algorithm is problematic; that seems like a code review issue 😛

Reply:
IMO unit tests should assert behavior and I feel like the hashing algorithm falls under that, although you do raise a good point about moving it to the `IPersonRepository`, actually moving the hashing logic feels like a better place than within the endpoint logic. I guess for now we can leave it as it is and spawn an offshoot ticket for updating the `IPersonRepository`?

Reply:
Sounds good to me.